libraries/package-deps-hash/src/getPackageDeps.ts

```diff
@@ -70,18 +71,34 @@ export function parseGitStatus(output: string, packagePath: string): Map<string,
        *   - '??' == untracked
        *   - 'R' == rename
        *   - 'RM' == rename with modifications
-       * filenames == path to the file, or files in the case of files that have been renamed
+       * filenames == path to the file, or files in the case of files that have been renamed. When filename characters
+       * are escaped, filenames will be surrounded by double-quotes. Some systems allow double-quotes in the
+       * filename as well, though these will be escaped and should be included by the regex. We will also keep the
+       * trailing whitespace after unquoted matches so that it can be used when reconstructing the filename
        */
-      const [changeType, ...filenames]: string[] = line
-        .trim()
-        .split(' ')
-        .filter((linePart) => !!linePart);
+      const match: RegExpMatchArray | null = line.match(/("(\\"|[^"])+")|(\S+\s*)/g);
+
+      if (match && match.length > 1) {
+        // Trim off leading and trailing double-quotes
+        const [changeType, ...filenames] = match.map((x) =>
+          x.match(/^".+"$/) ? x.slice(1, x.length - 1) : x
```

Comment:
This won't work for a string with `\"` escapes.

Overall this code would be better if moved into a function like:
```ts
/**
 * Extracts a string that was encoded by [spec for whatever the encoding mechanism
 * we believe is at play here].
 *
 * Examples
 * - `hello  ` --> `hello  `
 * - `"hello, \"world\""` --> `hello, "world"`
 * - `"hello, world"   ` --> `hello, world`
 * - `\\xy\n` --> `\xyn`
 */
private static _unencodeGitString(line: string): string {
}
```
I'm kind of curious what that spec is actually.